source/jsonParsing.ml

```diff
@@ -64,3 +64,201 @@ let optional_list_member ~f name json =
 let string_list_member = list_member ~f:to_string
 
 let path_list_member = list_member ~f:to_path
+
+module Location = struct
+  type t = {
+    start_line: int;
+    end_line: int;
+    start_column: int;
+    end_column: int;
+  }
+  [@@deriving show, eq]
+
+  let from_decoded_range range =
+    let (start_line, end_line), (start_column, end_column) = range in
+    { start_line; end_line; start_column; end_column }
+end
+
+module Node = struct
+  type 'a t = {
+    location: Location.t option;
```

Comment:
Why is the location optional? We shouldn't have empty locations when parsing json.
If it's just to create "dummy" nodes you can use a default location with line -1, column -1?

Reply:
it was for the dummy null nodes -> Yes I will use -1 -1 then :)